### PR TITLE
port ops to JSON (list insde)

### PR DESCRIPTION
**Pull request by bartlomieju**
* [x] `metrics.rs`
* [x] `net.ts`
* [x] `performance.rs`
* [x] `permissions.rs`
* [x] `process.rs`
* [x] `random.rs`
* [x] `repl.rs`
* [x] `resources.rs`
* [x] `timers.rs`
* [x] `workers.rs`

### CODE REVIEW

**Comment by bartlomieju** [comment]
@ry ready for review as well 😅 This PR should be landed after #2804 to adjust op ids properly

**Comment by bartlomieju** [comment]
Waiting for #2804

File: cli/ops/repl.rs [hunk 1]
```diff
@@ -1,78 +1,51 @@
...
+  _zero_copy: Option<PinnedBuf>,
+) -> Result<JsonOp, ErrBox> {
+  let args: ReplReadlineArgs = serde_json::from_value(args)?;
+  let rid = args.rid;
+  let prompt = args.prompt;
   debug!("op_repl_readline {} {}", rid, prompt);
 
-  blocking(base.sync(), move || {
-    let repl = resources::get_repl(rid)?;
+  // TODO(bartlomieju): this is a sync operation, why is it async in repl.ts?
```
Comment (ry) [inline cli/ops/repl.rs]: readline could potentially be async...

**Comment by bartlomieju** [comment]
Merged `master` and ported `metrics.rs` as well

File: cli/ops/repl.rs (hunk 1 above)
Comment (bartlomieju) [reply cli/ops/repl.rs]: Fair enough, should I rephrase the comment?

Comment (ry) [reply cli/ops/repl.rs]: Yes please

Comment (bartlomieju) [reply cli/ops/repl.rs]: Done

**Pull request merged**